src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/ValidatorFactory.cs

```diff
@@ -42,10 +50,51 @@ public IValidator Create(Type validatorType)
                     $"{nameof(IDictionary<FeedType, SourceRepository>)} and a {loggerType.Name}!");
             }
 
-            return constructor.Invoke(new object[] 
+            /// Get the <see cref="ServiceIndexEntry"/> prefixes requested by this <see cref="IValidator"/>.
+            _indexHijacker.BeginGetUsedEntryPrefixes();
+            Construct(constructor);
+            var prefixes = _indexHijacker.EndGetUsedEntryPrefixes();
+
+            /// Iterate through all possible combinations of <see cref="ServiceIndexEntry"/>s that were requested.
+            var serviceIndexEntryEqualityComparer = new ServiceIndexEntryEqualityComparer();
+            var possibleCombinations = 
+                prefixes
+                    .Select(p => 
+                        new KeyValuePair<string, IEnumerable<ServiceIndexEntry>>(p, 
+                            _indexHijacker
+                                .GetServiceIndexEntriesByTypePrefix(p)
+                                    .Distinct(serviceIndexEntryEqualityComparer)));
+
+            IList<IValidator> validators = new List<IValidator>();
+
+            var combinations = CombinationUtils.Enumerate(possibleCombinations);
+            if (combinations != null && combinations.Count() > 0)
```

Comment:
Unfortunately no. I am trying to get every possible combination of service index entries from a set.

E.g. I have several "buckets" of possible service index entries of each type.

[prefix1 -> a, b]
[prefix2 -> c, d]
[prefix3 -> e, f]

The possible combinations I want to find are
[a, c, e]
[a, c, f]
[a, d, e]
[a, d, f]
[b, c, e]
[b, c, f]
[b, d, e]
[b, d, f]

`SelectMany` on its own doesn't solve this problem, and, in fact, [I use `SelectMany` inside `CombinationUtils`.](https://github.com/NuGet/NuGet.Services.Metadata/pull/173/files#diff-1c9f47df766e28f4f48b81bc18d91b58R34)